cli/pcluster/config/config_patch.py

```diff
@@ -0,0 +1,256 @@
+        Get the max update policy level of the ConfigPatch.
+
+        This method provides a static indication of the actual applicability of the patch and it's meant for testing
+        purposes. The real applicability of a patch, which involves dynamic condition checks, must be performed
+        by calling the check() method.
+        """
+        return (
+            max(change.update_policy.level for change in self.changes)
+            if len(self.changes)
+            else UpdatePolicy.SUPPORTED.level
+        )
+
+    def _create_default_section(self, config, section):
+        """
+        Create a section of same type of the provided section, with all default parameter values.
+
+        The purpose of this operation is to allow sections comparison when a section is missing in base or target
+        configuration.
+
+        :param config: The configuration lacking the section
+        :param section: The section to create a default copy of
+        """
+        section_definition = section.definition
+        section_type = section_definition.get("type")
+        default_section = section_type(
+            section_definition=section_definition, pcluster_config=config, section_label=section.label
+        )
+
+        return default_section
+
+    def _get_config_section(self, config, section):
+        """
+        Get the section corresponding to the provided one in the specified configuration.
+
+        If no section is found, a default copy of the provided section is created on the fly.
+
+        :param config: The configuration where to get the section from
+        :param section: The section to be matched
+        :return: The matching section
+        """
+        config_section = config.get_section(section.key, section.label)
+
+        # If section is not present in base config, a default copy is generated
+        if not config_section:
+            config_section = self._create_default_section(config, section)
+        return config_section
+
+    def check(self):
+        """
+        Check the patch against the existing cluster stack.
+
+        All changes in the patch are checked against the existing cluster; their conditions are verified and a detailed
+        report is generated. Each line of the report will contain all the details about the detected change, together
+        with the corresponding reason if the change is not applicable and any action needed to unlock the problem.
+
+        :return A tuple containing the patch applicability and the report rows.
+        """
+        rows = [["section", "parameter", "old value", "new value", "check", "reason", "action_needed"]]
+
+        patch_allowed = True
+
+        for change in self.changes:
+            if change.update_policy != UpdatePolicy.IGNORED:
+                section_name = get_file_section_name(change.section_key, change.section_label)
+
+                check_result, reason, action_needed = change.update_policy.check(change, self)
```

Comment:
We're passing the `change` object to the `check` method of the `update_policy` of a specific `change`.

**change** .update_policy.check( **change** , self)

Is it possible to move the check method in the change itself?
I mean we should check a change not a policy.
Then the change can use its update policy to call the right checker.

Let's discuss if the comment is not clear.

Reply:
We could add a check() method to the Change Object, but it would still have to call the check() method of the UpdatePolicy (passing a self reference) since the logic to do the check is there (by definition, since it represents the policy).